### PR TITLE
Expand redis cache implementations

### PR DESCRIPTION
**Pull request by tylerhutcherson**
Introduces a semantic cache (backed by Redis, implemented in LangChain) as a viable option to improve application throughput.

More testing needed here.

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def startup_qna_backend():
             st.session_state['question'] = question
             with st.spinner("OpenAI and Redis are working to answer your question..."):
                 result = qna_chain({"query": question})
-                print(result, flush=True)
+                # print(result, flush=True)
```
Comment (Spartee) [inline app/app.py]: Remove?

File: app/qna/llm.py [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+# Env Vars and constants
+CACHE_TYPE = os.getenv("CACHE_TYPE")
 REDIS_HOST = os.getenv("REDIS_HOST", "localhost")
 REDIS_PORT = os.getenv("REDIS_PORT", 6379)
 REDIS_PASSWORD = os.getenv("REDIS_PASSWORD", "")
 REDIS_URL = f"redis://:{REDIS_PASSWORD}@{REDIS_HOST}:{REDIS_PORT}"
-
-INDEX_NAME = "wiki"
-
 OPENAI_API_TYPE = os.getenv("OPENAI_API_TYPE", "")
```
Comment (Spartee) [inline app/qna/llm.py]: default here?

**Review by Spartee — COMMENTED** [review]
A few comments, but this looks good to me.

File: app/qna/llm.py (hunk 2 above)
Comment (tylerhutcherson) [reply app/qna/llm.py]: adding

**Pull request merged**